src/hooks/useProposal/useProposal.ts

```diff
@@ -301,16 +306,36 @@ export function useProposal(proposal: ProposalTypes.Struct) {
         );
       }
       if ("xrpl" in requiredNamespaces || "xrpl" in optionalNamespaces) {
-        supportedNamespaces[SupportedNamespace.XRPL] =
-        buildXrpNamespace(requiredNamespaces, optionalNamespaces);
+        supportedNamespaces[SupportedNamespace.XRPL] = buildXrpNamespace(
+          requiredNamespaces,
+          optionalNamespaces,
+        );
       }
       return supportedNamespaces;
     },
-    [buildBip122Namespace, buildEip155Namespace, buildMvxNamespace, buildXrpNamespace],
+    [
+      buildBip122Namespace,
+      buildEip155Namespace,
+      buildMvxNamespace,
+      buildXrpNamespace,
+    ],
   );
 
   const sessionsQueryFn = useSessionsQueryFn(web3wallet);
 
+  const getValidation = useCallback((): ValidationStatus => {
+    return (proposal.proposer.metadata.verifyUrl ??
+      web3wallet.core.pairing
+        .getPairings()
+        .find((pairing) => pairing.topic === proposal.pairingTopic)
+        ?.peerMetadata?.verifyUrl ??
+      "UNKNOWN") as ValidationStatus;
+  }, [
+    proposal.pairingTopic,
+    proposal.proposer.metadata.verifyUrl,
+    web3wallet.core.pairing,
+  ]);
```

Comment:
I think it might be better to have a specific hook for this instead of sticking this in `useProposal` as this hook is made for another route and your function works in isolation